gpt_explainer/gpt_slide_expander.py

```diff
@@ -0,0 +1,272 @@
+                               f" relevant details. make the explanation up tp 3 sentences"
+                }
+            ]
+
+            gpt_response = openai.ChatCompletion.create(
+                model="gpt-3.5-turbo",
+                messages=messages
+            )
+
+            return gpt_response['choices'][0]['message']['content']
+        else:
+            return 'No text in the slide - unable to generate explanation'
+
+    @staticmethod
+    async def process_individual_slide_explanation_safely(slide_text: str, presentation_topic: str, max_retry: int,
+                                                          slide_index: int) -> str:
+        """
+        Process an individual slide task for generating an explanation.
+        Args:
+            slide_text: The text of the slide to generate an explanation for.
+            presentation_topic: The topic or context of the presentation.
+            max_retry: The maximum number of retries for generating the explanation.
+            slide_index: The index of the slide being processed.
+        Returns:
+            The generated explanation for the slide, or an error message if an `openai.error` occurs.
+        Raises:
+            openai.error: If an OpenAI API error occurs during explanation generation for the slide.
+        """
+        try:
+            return await GptSlideExpander.generate_explanation_for_slide_with_retry(slide_text, presentation_topic,
+                                                                                    max_retry)
+        except openai.error as e:
+            return f"ERROR - explanation generation for slide {slide_index} failed: {str(e)}"
+
+    def run_explainer(self) -> None:
+        """
+        Runs the GPT explainer system indefinitely, processing files dropped into the 'uploads' folder.
+        The explainer sleeps for 10 seconds between iterations.
+        """
+        while True:
+            files_to_process = self.get_files_to_process()
+            for file_path in files_to_process:
+                self.process_file(file_path)
+                time.sleep(10)  # Sleep for 10 seconds between processing each file
+
+    def get_files_to_process(self) -> List[str]:
+        """
+        Scans the 'uploads' folder and identifies the files that haven't been processed yet.
+        Returns:
+            list: A list of file paths that need to be processed.
+        """
+
+        files_to_process = []
+
+        with self.processed_files_lock:
+            processed_files = set(self.expanded_slide_explanations.keys())
+
+        for file_name in os.listdir(self.uploads_folder):
+            if file_name not in processed_files:
+                file_path = os.path.join(self.uploads_folder, file_name)
+                files_to_process.append(file_path)
+
+        return files_to_process
+
+    def process_file(self, file_path: str) -> None:
```

Comment:
Separate code to small functions.  Read more: https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be